src/libOpenImageIO/imageioplugin.cpp

```diff
@@ -537,6 +537,28 @@ ImageOutput::create(string_view filename, Filesystem::IOProxy* ioproxy,
         format = filename;
     }
 
+    ustring comma_sep_allowed_output_formats;
+    if (OIIO::getattribute("allowed_output_formats", TypeString,
+                           &comma_sep_allowed_output_formats)) {
+        std::vector<std::string> allowed_output_formats;
+        Strutil::split(comma_sep_allowed_output_formats, allowed_output_formats,
+                       ",", -1);
+
+        bool isValidFormat = 0;
+        for (std::string allowed_format : allowed_output_formats) {
```

Comment:
To solve the "empty list" problem and also eliminate all these string allocations and copies, I would suggest this change here:

```suggestion
    auto allowed_output_formats = Strutil::splitsv(OIIO::get_string_attribute("allowed_output_formats"), ",", -1);
    if (allowed_output_formats.size()) {
        bool isValidFormat = 0;
        for (auto allowed_format : allowed_output_formats) {
```